examples/ShowPlayer.cpp

```diff
@@ -270,8 +273,26 @@ void ShowPlayer::SendEntry(const ShowEntry &entry) {
   SendFrame(entry);
   m_playback_pos += entry.next_wait;
 
+  unsigned int timeout = entry.next_wait;
+  if (!m_simulate) {
+    // Using int64_t for target_delta here because
+    // we have to lose 1 bit anyway as InMilliSeconds() returns a signed
+    // 64-bit integer.
+    ola::TimeStamp now;
+    m_clock.CurrentMonotonicTime(&now);
+    int64_t target_delta = m_playback_pos - m_start_playback_pos;
+    int64_t current_delta = (now - m_start_ts).InMilliSeconds();
+    int64_t delay = target_delta - current_delta;
+    if (delay < 0) {
+      OLA_WARN << "Frame at line " << m_loader.GetCurrentLineNumber()
+               << " was meant to have completed " << -delay << " ms ago."
+               << " System too slow?";
+      delay = 0;
+    }
+    timeout = delay;
```

Comment:
it might, if UINT_MAX is smaller than INT64_MAX.

Though that would only happen if these two conditions are met:

- The duration of the current frame is near UINT_MAX ms.
- The current frame was emitted earlier so that the delay exceeds UINT_MAX ms.

If we assume a 32-bit unsigned int then it'd take a frame of ~49.7 days for that to happen.

But if it does happen then the value would just have its upper 32-bits truncated, resulting in a shorter delay than calculated.

I didn't add in a check for that because I didn't think anyone would craft that kind of input, but I can put one in to clamp `delay` to UINT_MAX.

Reply:
Thanks, as you say, hopefully unnecessary.